File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class WeekBucket(BaseModel):
    week: str
    leads: int
    qualified: int


class SourceSlice(BaseModel):
    name: str
    value: int


class StatusCount(BaseModel):
    status: str
    count: int


class AnalyticsOverview(BaseModel):
    total_leads: int
    qualified_leads: int
    avg_score: float
    leads_over_time: list[WeekBucket]
    source_split: list[SourceSlice]
    status_breakdown: list[StatusCount]

File: src/lib/api/analytics.ts
import { apiFetch } from './client';

export interface WeekBucketDTO {
  week: string;
  leads: number;
  qualified: number;
}

export interface SourceSliceDTO {
  name: string;
  value: number;
}

export interface StatusCountDTO {
  status: string;
  count: number;
}

export interface AnalyticsOverviewDTO {
  total_leads: number;
  qualified_leads: number;
  avg_score: number;
  leads_over_time: WeekBucketDTO[];
  source_split: SourceSliceDTO[];
  status_breakdown: StatusCountDTO[];
}

export function getAnalyticsOverview(token: string | null) {
  return apiFetch<AnalyticsOverviewDTO>('/api/v1/analytics/overview', token);
}
